pytest_spec/patch.py

```diff
@@ -42,8 +43,18 @@ def _is_nodeid_has_test(nodeid):
     return False
 
 
-def _get_test_path(nodeid):
-    return nodeid.rsplit("::", 1)[0]
+def _get_test_path(nodeid, header):
+    levels = nodeid.split("::")
+
+    if len(levels) > 2:
+        class_name = levels[1]
+        test_case = _split_words(_remove_class_prefix(class_name))
+    else:
+        module_name = os.path.split(levels[0])[1]
+        class_name = ''
+        test_case = _capitalize_first_letter(_replace_underscores(_remove_test_prefix(_remove_file_extension(module_name))))
```

Comment:
I don't like such nested calls but this is something to re-think (not for this change)